Source/DafnyCore/Backends/Dafny/AST.dfy

```diff
@@ -130,10 +130,30 @@ module {:extern "DAST"} DAST {
 
   datatype Primitive = Int | Real | String | Bool | Char | Native
 
-  // USIZE is for whatever target considers that native arrays can be indexed with
+
   datatype NewtypeRange =
-    | U8 | I8 | U16 | I16 | U32 | I32 | U64 | I64 | U128 | I128 | USIZE | BigInt
```

Comment:
Given that this is in a file AST, does it conceptually mirror the Dafny C# AST?

Reply:
Almost, it's a desugared version of the C# AST, the intermediate representation. For example, comprehension expressions have disappeared, no distinction between functions and methods, and subset types become type synonyms.